Changed the Functionality of Delete all Button to delete selected Boo…

harsh551 opened the pull request:
…kmarks

<!-- Please replace #XX below with an existing issue number. Remove the line entirely if none exist. -->
Fixes #161 

**Description**
<!-- Add your description below. -->

**Other information**
<!-- Add any other information below or delete the "Other Information" line entirely. -->
Hey, @makkoncept I have Changed the code as you want. I think now it's done.
![1](https://user-images.githubusercontent.com/43513378/76081342-be729900-5f5d-11ea-9b75-88ff839dbee8.png)
![2](https://user-images.githubusercontent.com/43513378/76081398-df3aee80-5f5d-11ea-9a33-5ed62da3d6d6.png)
![3](https://user-images.githubusercontent.com/43513378/76081459-01347100-5f5e-11ea-9313-317139dffa19.png)


**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

<!-- Make sure you read and understand the following attestation. -->

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Commit:
Changed the Functionality of Delete all Button to delete selected Bookmarks

Commit:
Re formatted code ...

makkoncept commented on src/popup/bookmarkModule.js:
```diff
@@ -302,18 +302,32 @@ document.addEventListener('DOMContentLoaded', () => {
+          addSpinner(elements.spinnerPlaceholderBookmarks, 'original');
+          removeOldSearchResults();
+          removeLoaderAnimation();
+          restoreInitialContent('primary');
```
The following function calls do not do anything important to achieve the current behavior. They are redundant.
- `removeOldSearchResults()`
- `removeLoaderAnimation()`
-  `restoreInitialContent('primary')`

But I would argue for a better solution.

**Current workflow**
Right now, even if the user deletes a single image, the whole bookmarks section is loaded again. Image having a very large number of images in this section. Reloading them after deleting a few images (relatively speaking) would lead to triggering 99% of the same API requests and consecutively loading the image thumbnails. Add to that a slow internet connection or a low-end computer. 

**Improvement**
We can take the help of the [masonry grid](https://masonry.desandro.com/) used by the extension. It has the capability to reorganize the images in the grid after some of them are deleted without loading the images again ([check this](https://codepen.io/desandro/pen/IBJeq)). This is exactly what we want. In fact, this property is used to remove the bookmarks when the user clicks on the bookmark icon on the images in the bookmarks section ([implemented here](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/src/popup/bookmarkModule.js#L45))

In short: Instead of reloading the images again after deleting, remove the selected images from the DOM and let the masonry grid does the heavy lifting of reorganizing the rest of the images.

Commit:
Changed the deletebookmarks eventlisner as directed

Commit:
Reformatted code

harsh551 commented:
I think @makkoncept that's what you are saying about.
Removed redundant functions


- `removeOldSearchResults()`
- `removeLoaderAnimation()`
- `restoreInitialContent('primary')`

added 

- `msnry.layout()`

Thank you for guiding me about this msnry grid and helping me to understand the workflow in a better way.

makkoncept commented:
Have you tried it on your system? Is it working as expected?

harsh551 commented:
Yes

makkoncept commented on src/popup/bookmarkModule.js:
```diff
@@ -302,18 +302,29 @@ document.addEventListener('DOMContentLoaded', () => {
           removeBookmarkImages();
           msnry.layout();
-          restoreInitialContent('bookmarks');
+          loadImages();
```
@harsh551 What you have done is remove the loading spinner, therefore it looks like the images are not loaded again. But, since you are using the `loadImages()` function, they are indeed being reloaded after every deletion.  As I have said above, we have to implement something like [this](https://codepen.io/desandro/pen/IBJeq) (i.e don't reload the images after deletion but let the masonry grid reorganize them). This is what happens when you remove the images by clicking the bookmark icon (gif below). [This function](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/src/popup/bookmarkModule.js#L45) implements that.

![ezgif com-crop (2)](https://user-images.githubusercontent.com/34679965/76159894-00acfd80-614b-11ea-98ba-1bf017774170.gif)

makkoncept reviewed (approved):
@harsh551 I guess we can move the problem of reloading of images after deletion  to another issue. This PR solves the original issue well. Merging.
